tests/test_html_formatter.py::test_ctags fails when `python-ctags3` is installed

When the `python-ctags3` package is installed, the test suite fails:

```pytb
============================================================== FAILURES ===============================================================
_____________________________________________________________ test_ctags ______________________________________________________________

    def test_ctags():
        try:
            import ctags # noqa: F401
        except ImportError:
            # we can't check without the ctags module, but at least check the exception
            assert pytest.raises(
                RuntimeError, HtmlFormatter, tagsfile='support/tags'
            )
        else:
            # this tagfile says that test_ctags() is on line 165, even if it isn't
            # anymore in the actual source
>           fmt = HtmlFormatter(tagsfile='support/tags', lineanchors='L',
                                tagurlformat='%(fname)s%(fext)s')

tests/test_html_formatter.py:243: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pygments/formatters/html.py:432: in __init__
    self._ctags = ctags.CTags(self.tagsfile)
src/_readtags.pyx:91: in _readtags.CTags.__cinit__
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   Exception: Invalid tag file

src/_readtags.pyx:131: Exception
```

To reproduce:

```
tox e -e py311 -- pip install python-ctags3
tox r -e py311
```

Full test log: [test.log](https://github.com/pygments/pygments/files/12271618/test.log)

Reproduced on Python 3.11.4, Gentoo Linux amd64, git @ 9162c079eac12bbf17b187013ab0c46efb7eb43f (i.e. 2.16.1).

--- a/pygments/formatters/html.py
+++ b/pygments/formatters/html.py
@@ -323,6 +323,7 @@ class HtmlFormatter(Formatter):
         If set to the path of a ctags file, wrap names in anchor tags that
         link to their definitions. `lineanchors` should be used, and the
         tags file should specify line numbers (see the `-n` option to ctags).
+        The tags file is assumed to be encoded in UTF-8.
 
         .. versionadded:: 1.6
 
@@ -908,7 +909,7 @@ class HtmlFormatter(Formatter):
     def _lookup_ctag(self, token):
         entry = ctags.TagEntry()
         if self._ctags.find(entry, token.encode(), 0):
-            return entry['file'], entry['lineNumber']
+            return entry['file'].decode(), entry['lineNumber']
         else:
             return None, None
 

--- a/tests/test_html_formatter.py
+++ b/tests/test_html_formatter.py
@@ -235,12 +235,12 @@ def test_ctags():
     except ImportError:
         # we can't check without the ctags module, but at least check the exception
         assert pytest.raises(
-            RuntimeError, HtmlFormatter, tagsfile='support/tags'
+            RuntimeError, HtmlFormatter, tagsfile='tests/support/tags'
         )
     else:
         # this tagfile says that test_ctags() is on line 165, even if it isn't
         # anymore in the actual source
-        fmt = HtmlFormatter(tagsfile='support/tags', lineanchors='L',
+        fmt = HtmlFormatter(tagsfile='tests/support/tags', lineanchors='L',
                             tagurlformat='%(fname)s%(fext)s')
         outfile = StringIO()
         fmt.format(tokensource, outfile)

